Modules/_sqlite/cursor.c

```diff
@@ -533,7 +533,7 @@ _pysqlite_query_execute(pysqlite_Cursor* self, int multiple, PyObject* operation
         goto error;
     }
 
-    if (self->statement->in_use) {
+    if (sqlite3_stmt_busy(self->statement->st)) {
```

Comment:
`sqlite3_stmt_busy` also does a NULL pointer check, but we'll avoid a function call if we do it ourselves. Resetting a statement is a frequent operation.